google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta2/StreamWriterV2.java

```diff
@@ -171,8 +196,29 @@ public void run() {
                         "Stream is closed due to " + connectionFinalStatus.toString())));
         return requestWrapper.appendResult;
       }
+
+      ++this.inflightRequests;
+      this.inflightBytes += requestWrapper.messageSize;
       waitingRequestQueue.addLast(requestWrapper);
```

Comment:
Why do you want to check the limit after adding the request to wait queue?

Reply:
We could do it either way. Bigang mentioned to do it AFTER, so I just followed it to avoid conversions like this...

Reply:
It's okay to check after adding the queue. Holding the Append() is a natural way to throttle the user when necessary.